lib/cache/collections.go

```diff
@@ -1246,9 +1246,20 @@ func (databaseExecutor) getAll(ctx context.Context, cache *Cache, loadSecrets bo
 
 func (databaseExecutor) upsert(ctx context.Context, cache *Cache, resource types.Database) error {
 	if err := cache.databasesCache.CreateDatabase(ctx, resource); err != nil {
+		// Skip databases that failed to be validated to avoid stopping cache
+		// backend events consumption (which can lead to an invalid cache
+		// state). This case can happen during Teleport upgrades/downgrades,
+		// where the database stored on the backend is no longer valid given the
+		// Teleport version.
+		if trace.IsBadParameter(err) {
+			cache.Logger.Infof("Adding database %q to cache was skipped due to validation errors. Please update your database spec accordingly: %s", resource.GetName(), err)
```

Comment:
Cloud users don't have access to their auth logs, and most on-prem users are not staring at their auth server logs to notice something like this.